core/server/src/main/java/alluxio/proxy/FileSystemClientRestServiceHandler.java

```diff
@@ -0,0 +1,474 @@
+        mFileSystem.rename(new AlluxioURI(srcPath), new AlluxioURI(dstPath));
+        return null;
+      }
+    });
+  }
+
+  /**
+   * @summary set an attribute
+   * @param path the file path
+   * @param pinned the pinned flag value to use
+   * @param ttl the time-to-live (in seconds) to use
+   * @param persisted the persisted flag value to use
+   * @param owner the file owner
+   * @param group the file group
+   * @param mode the file permission bits
+   * @param recursive whether the attribute should be set recursively
+   * @param ttlAction action to take after TTL is expired
+   * @return the response object
+   */
+  @POST
+  @Path(SET_ATTRIBUTE)
+  @ReturnType("java.lang.Void")
+  public Response setAttribute(@QueryParam("path") final String path,
+      @QueryParam("pinned") final Boolean pinned, @QueryParam("ttl") final Long ttl,
+      @QueryParam("persisted") final Boolean persisted, @QueryParam("owner") final String owner,
+      @QueryParam("group") final String group, @QueryParam("mode") final Short mode,
+      @QueryParam("recursive") final Boolean recursive,
+      @QueryParam("ttxAction") final TtlAction ttlAction) {
+    return RestUtils.call(new RestUtils.RestCallable<Void>() {
+      @Override
+      public Void call() throws Exception {
+        SetAttributeOptions options = SetAttributeOptions.defaults();
+        Preconditions.checkNotNull(path, "required 'path' parameter is missing");
+        if (persisted != null) {
+          options.setPersisted(persisted);
+        }
+        if (pinned != null) {
+          options.setPinned(pinned);
+        }
+        if (group != null) {
+          options.setGroup(group);
+        }
+        if (owner != null) {
+          options.setOwner(owner);
+        }
+        if (mode != null) {
+          options.setMode(mode);
+        }
+        if (recursive != null) {
+          options.setRecursive(recursive);
+        }
+        if (ttl != null) {
+          options.setTtl(ttl);
+        }
+        if (ttlAction != null) {
+          options.setTtlAction(ttlAction);
+        }
+        mFileSystem.setAttribute(new AlluxioURI(path), options);
+        return null;
+      }
+    });
+  }
+
+  /**
+   * @summary unmount a path
+   * @param path the file path
+   * @return the response object
+   */
+  @POST
+  @Path(UNMOUNT)
+  @ReturnType("java.lang.Void")
+  public Response unmount(@QueryParam("path") final String path) {
+    return RestUtils.call(new RestUtils.RestCallable<Void>() {
+      @Override
+      public Void call() throws Exception {
+        Preconditions.checkNotNull(path, "required 'path' parameter is missing");
+        mFileSystem.unmount(new AlluxioURI(path));
+        return null;
+      }
+    });
+  }
+
+  /**
+   * @summary upload a file
+   * @param path the file path
+   * @param blockSizeBytes the target block size in bytes
+   * @param mode the octal mode to used for file permissions
+   * @param persisted whether directory should be persisted
```

Comment:
I also went through all the options and added ones that were missing (e.g. location policies)